src/main/java/seedu/address/model/task/Time.java

```diff
@@ -1,42 +1,84 @@
 package seedu.address.model.task;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.util.Locale;
+
 import seedu.address.commons.exceptions.IllegalValueException;
 
 /**
- * Represents a Task's phone number in the SmartyDo.
- * Guarantees: immutable; is valid as declared in {@link #isValidPhone(String)}
+ * Represents a Task's Date in the SmartyDo.
+ * Guarantees: immutable; is valid as declared in {@link #isValidDate(String)}
  */
 public class Time {
 
-    public static final String MESSAGE_PHONE_CONSTRAINTS = "Task phone numbers should only contain numbers";
-    public static final String PHONE_VALIDATION_REGEX = "\\d{4}";
+    public static final String MESSAGE_DATE_CONSTRAINTS = "Task Dates should be in valid UK-format "
+            + "DD/MMM/YYYY or DD/MM/YYYY or DD.MM.YYYY or DD.MMM.YYY or DD-MM-YYYY or DD-MMM-YYYY";
+    /**
+     * Date validation in UK format, includes checks for valid date during leap years.
+     * Supported Formats: dd/mmm/yyyy, dd-mmm-yyyy, dd/mm/yyyy, dd-mm-yyyy, dd.mm.yyyy
+     *
+     * @author Alok Chaudhary, Filbert(with the help of Java regex converter)
+     */
+    public static final String DATE_VALIDATION_REGEX = "^(?:(?:31(\\/|-|\\.)(?:0?[13578]|1[02]|"
+            + "(?:Jan|Mar|May|Jul|Aug|Oct|Dec)))\\1|(?:(?:29|30)(\\/|-|\\.)(?:0?[1,3-9]|1[0-2]|"
+            + "(?:Jan|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec))\\2))(?:(?:1[6-9]|[2-9]\\d)?\\d{2})$|^"
+            + "(?:29(\\/|-|\\.)(?:0?2|(?:Feb))\\3(?:(?:(?:1[6-9]|[2-9]\\d)?(?:0[48]|[2468][048]|[13579][26])|"
+            + "(?:(?:16|[2468][048]|[3579][26])00))))$|^(?:0?[1-9]|1\\d|2[0-8])(\\/|-|\\.)(?:(?:0?[1-9]|"
+            + "(?:Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep))|(?:1[0-2]|(?:Oct|Nov|Dec)))"
+            + "\\4(?:(?:1[6-9]|[2-9]\\d)?\\d{2})$";
```

Comment:
Wow I wanted to refactor this, then I don't even know where to start... hahas forget it.